ocs_ci/ocs/perftests.py

```diff
@@ -41,9 +42,61 @@ def setup(self):
         self.benchmark_obj = None  # place holder for the benchmark object
         self.client_pod = None  # Place holder for the client pod object
         self.dev_mode = config.RUN["cli_params"].get("dev_mode")
-        self.environment = get_environment_info()
         self.pod_obj = OCP(kind="pod")
 
+        # Collecting all Environment configuration Software & Hardware
+        # for the performance report.
+        self.environment = get_environment_info()
+        self.environment["clusterID"] = get_running_cluster_id()
+
+        self.get_osd_info()
+
+        self.get_node_info(node_type="master")
+        self.get_node_info(node_type="worker")
+
+    def get_osd_info(self):
+        """
+        Getting the OSD's information and update the main environment
+        dictionary.
+
+        """
+        ct_pod = pod.get_ceph_tools_pod()
+        osd_info = ct_pod.exec_ceph_cmd(ceph_cmd="ceph osd df")
+        self.environment["osd_size"] = osd_info.get("nodes")[0].get("crush_weight")
+        self.environment["osd_num"] = len(osd_info.get("nodes"))
+        self.environment["total_capacity"] = osd_info.get("summary").get(
+            "total_kb_avail"
+        )
+        self.environment["ocs_nodes_num"] = len(node.get_ocs_nodes())
+
+    def get_node_info(self, node_type="master"):
+        """
+        Getting node type hardware information and update the main environment
+        dictionary.
+
+        Args:
+            node_type (str): the node type to collect data about,
+              can be : master / worker - the default is master
+
+        """
+        if node_type == "master":
+            nodes = node.get_master_nodes()
+        elif node_type == "worker":
```

Comment:
This is assuming that the worker nodes are OCS nodes which might not be the case for scaled nodes. We might also have different instance types for nodes running app pods and OCS pods. So, it would be a good idea to look for OCS-labeled nodes. 
This can be handled in a subsequent PR too, as we are not using different worker nodes.